app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, tuple(sorted(kwargs.items())))
+        if key in cached_results:
+            print("Getting from cache")
+            return cached_results[key]
+        else:
+            print("Calculating new result")
+            return cached_results[key] if key in cached_results else (
```

Comment:
cached_results[key] = result
```suggestion
            cached_results[key] = cached_results.setdefault(key, func(*args, **kwargs))
```